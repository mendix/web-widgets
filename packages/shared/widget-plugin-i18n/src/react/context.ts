import { createContext } from "react";
import { PluginI18n } from "../types";

const contextPath = "com.mendix.widget.plugin.i18n.context" as const;

type I18nContext = React.Context<PluginI18n | null>;

const createI18nContext = (): I18nContext => createContext<PluginI18n | null>(null);

declare global {
    interface Window {
        [contextPath]?: ReturnType<typeof createI18nContext>;
    }
}

export function getGlobalI18nPluginContext(): I18nContext {
    return window[contextPath] ?? createI18nContext();
}
