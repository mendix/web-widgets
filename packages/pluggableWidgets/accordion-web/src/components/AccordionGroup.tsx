import { createElement, KeyboardEvent, ReactElement, ReactNode, useCallback, useEffect, useRef, useState } from "react";
import classNames from "classnames";
import "../ui/accordion-main.scss";
import { LoadContentEnum } from "typings/AccordionProps";

/* eslint-disable no-unused-vars */
export const enum Target {
    FIRST = "first",
    LAST = "last",
    PREVIOUS = "previous",
    NEXT = "next"
}
/* eslint-enable no-unused-vars */

export type AccordionGroupIcon = { icon: ReactNode } | { expandIcon: ReactNode; collapseIcon: ReactNode };

export interface AccordionGroupProps {
    id: string;
    header: ReactNode;
    content: ReactNode;
    collapsed: boolean;
    visible: boolean;
    dynamicClassName?: string;
    collapsible: boolean;
    toggleCollapsed?: () => void;
    onToggleCompletion?: (collapsed: boolean) => void;
    changeFocus?: (focusedGroupHeader: EventTarget | null, focusTargetGroupHeader: Target) => void;
    animateContent?: boolean;
    generateHeaderIcon?: (collapsed: boolean) => ReactElement;
    showHeaderIcon?: "right" | "left" | "no";
    loadContent: LoadContentEnum;
}

export function AccordionGroup(props: AccordionGroupProps): ReactElement | null {
    const { animateContent, changeFocus, showHeaderIcon, toggleCollapsed, onToggleCompletion, loadContent } = props;

    const [renderCollapsed, setRenderCollapsed] = useState(props.collapsed);
    const previousRenderCollapsed = useRef(renderCollapsed);
    const animatingContent = useRef<boolean>(false);

    const rootRef = useRef<HTMLDivElement>(null);
    const contentWrapperRef = useRef<HTMLDivElement>(null);
    const contentRef = useRef<HTMLDivElement>(null);
    const renderContent = useRef(loadContent === "always");

    renderContent.current ||= !renderCollapsed;

    const completeTransitioning = useCallback((): void => {
        if (contentWrapperRef.current && rootRef.current && animatingContent.current) {
            if (!renderCollapsed) {
                rootRef.current.classList.add("widget-accordion-group-collapsed");
                rootRef.current.classList.remove("widget-accordion-group-collapsing");
                setRenderCollapsed(true);
            } else {
                rootRef.current.classList.remove("widget-accordion-group-expanding");
                contentWrapperRef.current.style.height = "";
                setRenderCollapsed(false);
            }

            animatingContent.current = false;
        }
    }, [renderCollapsed]);

    useEffect(() => {
        if (
            props.collapsed !== renderCollapsed &&
            rootRef.current &&
            contentWrapperRef.current &&
            contentRef.current &&
            animateContent
        ) {
            animatingContent.current = true;
            if (props.collapsed) {
                contentWrapperRef.current.style.height = `${contentRef.current.getBoundingClientRect().height}px`;
                rootRef.current.classList.add("widget-accordion-group-collapsing");

                setTimeout(() => {
                    if (contentWrapperRef.current) {
                        contentWrapperRef.current.style.height = "";
                    }
                }, 50);
            } else {
                rootRef.current.classList.add("widget-accordion-group-expanding");
                rootRef.current.classList.remove("widget-accordion-group-collapsed");

                setTimeout(() => {
                    if (contentWrapperRef.current && contentRef.current) {
                        contentWrapperRef.current.style.height = `${
                            contentRef.current.getBoundingClientRect().height
                        }px`;
                    }
                }, 50);
            }
        } else if (props.collapsed !== renderCollapsed && (!animateContent || !props.visible)) {
            setRenderCollapsed(props.collapsed);
        }
    }, [props.collapsed, props.visible, renderCollapsed, animateContent]);

    useEffect(() => {
        if (renderCollapsed !== previousRenderCollapsed.current) {
            previousRenderCollapsed.current = renderCollapsed;
            onToggleCompletion?.(props.collapsed);
        }
    }, [props.collapsed, onToggleCompletion, renderCollapsed]);

    const onKeydownHandler = useCallback(
        (event: KeyboardEvent<HTMLDivElement>) => {
            switch (event.key) {
                case "Enter":
                case " ":
                    event.preventDefault();
                    toggleCollapsed?.();
                    break;
                case "Home":
                    event.preventDefault();
                    changeFocus?.(event.currentTarget, Target.FIRST);
                    break;
                case "End":
                    event.preventDefault();
                    changeFocus?.(event.currentTarget, Target.LAST);
                    break;
                case "Up": // Microsoft Edge value
                case "ArrowUp":
                    event.preventDefault();
                    changeFocus?.(event.currentTarget, Target.PREVIOUS);
                    break;
                case "Down": // Microsoft Edge value
                case "ArrowDown":
                    event.preventDefault();
                    changeFocus?.(event.currentTarget, Target.NEXT);
                    break;
            }
        },
        [toggleCollapsed, changeFocus]
    );

    if (!props.visible) {
        return null;
    }

    return (
        <section
            ref={rootRef}
            className={classNames(
                "widget-accordion-group",
                {
                    "widget-accordion-group-collapsed": renderCollapsed
                },
                props.dynamicClassName
            )}
        >
            <header className={"widget-accordion-group-header"}>
                <div
                    id={`${props.id}HeaderButton`}
                    className={classNames("widget-accordion-group-header-button", {
                        "widget-accordion-group-header-button-clickable": props.collapsible,
                        "widget-accordion-group-header-button-icon-left":
                            props.collapsible && showHeaderIcon === "left",
                        "widget-accordion-group-header-button-icon-right":
                            props.collapsible && showHeaderIcon === "right"
                    })}
                    tabIndex={props.collapsible ? 0 : undefined}
                    data-focusindex={0}
                    role="button"
                    onClick={props.collapsible ? toggleCollapsed : undefined}
                    onKeyDown={props.collapsible ? onKeydownHandler : undefined}
                    aria-expanded={!renderCollapsed}
                    aria-disabled={!props.collapsible}
                    aria-controls={`${props.id}ContentWrapper`}
                >
                    {props.header}
                    {props.collapsible && (showHeaderIcon === "left" || showHeaderIcon === "right")
                        ? props.generateHeaderIcon?.(renderCollapsed ?? false)
                        : null}
                </div>
            </header>
            <div
                ref={contentWrapperRef}
                id={`${props.id}ContentWrapper`}
                className={"widget-accordion-group-content-wrapper"}
                data-focusindex={0}
                onTransitionEnd={completeTransitioning}
                role="region"
                aria-labelledby={`${props.id}HeaderButton`}
            >
                <div ref={contentRef} className={"widget-accordion-group-content"}>
                    {renderContent.current && props.content}
                </div>
            </div>
        </section>
    );
}
