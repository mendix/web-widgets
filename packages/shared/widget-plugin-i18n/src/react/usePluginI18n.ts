import { useState } from "react";
import { I18nController } from "../controllers/I18nController";
import { PluginI18n } from "../types";

export function usePluginI18n(languageCode: string): PluginI18n | null {
    const [plugin, setPlugin] = useState<PluginI18n | null>(() => {
        new I18nController({
            languageCode,
            onPluginChange: plugin => setPlugin(plugin)
        });

        return null;
    });

    return plugin;
}
