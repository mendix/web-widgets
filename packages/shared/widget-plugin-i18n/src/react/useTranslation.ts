import { useContext } from "react";
import { PluginI18n } from "../types";
import { getGlobalI18nPluginContext } from "./context";

export function useTranslation(): PluginI18n | null {
    return useContext(getGlobalI18nPluginContext());
}
