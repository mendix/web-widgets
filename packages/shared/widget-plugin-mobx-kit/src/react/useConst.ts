import { useRef } from "react";

export function useConst<T>(fn: T): T extends (...args: never) => infer R ? R : T {
    return (useRef<T | null>(null).current ??= typeof fn === "function" ? fn() : fn);
}
