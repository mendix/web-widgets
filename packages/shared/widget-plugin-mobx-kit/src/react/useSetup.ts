import { useEffect } from "react";
import { useConst } from "./useConst";

export function useSetup<
    T extends {
        setup(): void | (() => void);
    }
>(fn: () => T): T {
    const obj = useConst(fn);
    useEffect(() => obj.setup(), [obj]);
    return obj;
}
