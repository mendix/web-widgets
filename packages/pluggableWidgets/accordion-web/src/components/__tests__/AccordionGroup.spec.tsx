import { createElement } from "react";
import { mount, shallow, ShallowWrapper } from "enzyme";
import { AccordionGroup, AccordionGroupProps, Target } from "../AccordionGroup";

describe("AccordionGroup", () => {
    let defaultAccordionGroupProps: AccordionGroupProps;

    beforeEach(() => {
        defaultAccordionGroupProps = {
            id: "id",
            header: "header",
            content: <span>content</span>,
            collapsed: true,
            visible: true,
            dynamicClassName: "class-name",
            collapsible: false,
            animateContent: false, // testing animations with Enzyme doesn't work
            generateHeaderIcon: jest.fn(),
            showHeaderIcon: "right",
            loadContent: "always"
        };
    });

    it("doesn't render when the group isn't visible", () => {
        const accordionGroup = shallow(<AccordionGroup {...defaultAccordionGroupProps} visible={false} />);

        expect(defaultAccordionGroupProps.generateHeaderIcon).not.toHaveBeenCalled();
        expect(accordionGroup).toMatchSnapshot();
    });

    describe("collapsible", () => {
        function mountCollapsibleAccordionGroup(
            accordionGroupProps: AccordionGroupProps,
            toggleCollapsed?: () => void,
            changeFocus?: (focusedGroupHeader: EventTarget | null, focusTargetGroupHeader: Target) => void
        ): ShallowWrapper {
            const resToggleCollapsed = toggleCollapsed ?? jest.fn();

            return shallow(
                <AccordionGroup
                    {...accordionGroupProps}
                    collapsible
                    toggleCollapsed={resToggleCollapsed}
                    changeFocus={changeFocus}
                />
            );
        }

        it("renders correctly when the group is visible and collapsed", () => {
            const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps);

            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(true);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("renders correctly when the group is visible and expanded", () => {
            const accordionGroup = mountCollapsibleAccordionGroup({
                ...defaultAccordionGroupProps,
                collapsed: false
            });

            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(false);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("renders correctly when the group is visible and gets expanded", () => {
            const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(true);

            accordionGroup.setProps({ collapsed: false });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(3);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(false);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("renders correctly when the group is visible and gets collapsed", () => {
            const accordionGroup = mountCollapsibleAccordionGroup({
                ...defaultAccordionGroupProps,
                collapsed: false
            });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(false);

            accordionGroup.setProps({ collapsed: true });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(3);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(true);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("renders correctly when the group becomes visible and is collapsed", () => {
            const accordionGroup = mountCollapsibleAccordionGroup({
                ...defaultAccordionGroupProps,
                visible: false
            });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(0);

            accordionGroup.setProps({ visible: true });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(true);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("renders correctly when the group becomes visible and is expanded", () => {
            const accordionGroup = mountCollapsibleAccordionGroup({
                ...defaultAccordionGroupProps,
                collapsed: false,
                visible: false
            });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(0);

            accordionGroup.setProps({ visible: true });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(1);
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledWith(false);
            expect(accordionGroup).toMatchSnapshot();
        });

        describe("header", () => {
            it("calls toggleCollapsed when clicking to expand", () => {
                const toggleCollapsedMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps, toggleCollapsedMock);

                accordionGroup.find(".widget-accordion-group-header-button").simulate("click");
                expect(toggleCollapsedMock).toHaveBeenCalledTimes(1);
            });

            it("calls toggleCollapsed when clicking to collapse", () => {
                const toggleCollapsedMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(
                    { ...defaultAccordionGroupProps, collapsed: false },
                    toggleCollapsedMock
                );

                accordionGroup.find(".widget-accordion-group-header-button").simulate("click");
                expect(toggleCollapsedMock).toHaveBeenCalledTimes(1);
            });

            it("calls toggleCollapsed on space key down", () => {
                const toggleCollapsedMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps, toggleCollapsedMock);

                const keyboardEvent = { key: " ", preventDefault: jest.fn() };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(toggleCollapsedMock).toHaveBeenCalledTimes(1);
            });

            it("calls toggleCollapsed on Enter key down", () => {
                const toggleCollapsedMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps, toggleCollapsedMock);

                const keyboardEvent = { key: "Enter", preventDefault: jest.fn() };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(toggleCollapsedMock).toHaveBeenCalledTimes(1);
            });

            it("calls changeFocus on arrow down key down", () => {
                const changeFocusMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(
                    defaultAccordionGroupProps,
                    undefined,
                    changeFocusMock
                );

                const keyboardEvent = { key: "ArrowDown", preventDefault: jest.fn(), currentTarget: "currentTarget" };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledWith(keyboardEvent.currentTarget, Target.NEXT);
            });

            it("calls changeFocus on arrow up key down", () => {
                const changeFocusMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(
                    defaultAccordionGroupProps,
                    undefined,
                    changeFocusMock
                );

                const keyboardEvent = { key: "ArrowUp", preventDefault: jest.fn(), currentTarget: "currentTarget" };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledWith(keyboardEvent.currentTarget, Target.PREVIOUS);
            });

            it("calls changeFocus on Home key down", () => {
                const changeFocusMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(
                    defaultAccordionGroupProps,
                    undefined,
                    changeFocusMock
                );

                const keyboardEvent = { key: "Home", preventDefault: jest.fn(), currentTarget: "currentTarget" };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledWith(keyboardEvent.currentTarget, Target.FIRST);
            });

            it("calls changeFocus on End key down", () => {
                const changeFocusMock = jest.fn();

                const accordionGroup = mountCollapsibleAccordionGroup(
                    defaultAccordionGroupProps,
                    undefined,
                    changeFocusMock
                );

                const keyboardEvent = { key: "End", preventDefault: jest.fn(), currentTarget: "currentTarget" };

                accordionGroup.find(".widget-accordion-group-header-button").simulate("keydown", keyboardEvent);
                expect(keyboardEvent.preventDefault).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledTimes(1);
                expect(changeFocusMock).toHaveBeenCalledWith(keyboardEvent.currentTarget, Target.LAST);
            });

            it("calls onToggleCompletion", () => {
                const onToggleCompletionMock = jest.fn();

                const accordionGroup = mount(
                    <AccordionGroup
                        {...defaultAccordionGroupProps}
                        collapsible
                        onToggleCompletion={onToggleCompletionMock}
                    />
                );

                accordionGroup.setProps({ collapsed: false });
                expect(onToggleCompletionMock).toHaveBeenCalledTimes(1);
                expect(onToggleCompletionMock).toHaveBeenCalledWith(false);
            });

            it("applies the correct class when the header icon is aligned right", () => {
                const accordionGroup = mountCollapsibleAccordionGroup(defaultAccordionGroupProps);

                expect(accordionGroup.find(".widget-accordion-group-header-button").prop("className")).toContain(
                    "widget-accordion-group-header-button-icon-right"
                );
            });

            it("applies the correct class when the header icon is aligned left", () => {
                const accordionGroup = mountCollapsibleAccordionGroup({
                    ...defaultAccordionGroupProps,
                    showHeaderIcon: "left"
                });

                expect(accordionGroup.find(".widget-accordion-group-header-button").prop("className")).toContain(
                    "widget-accordion-group-header-button-icon-left"
                );
            });

            it("doesn't render the icon when set to not visible", () => {
                const accordionGroup = mountCollapsibleAccordionGroup({
                    ...defaultAccordionGroupProps,
                    showHeaderIcon: "no"
                });

                expect(accordionGroup).toMatchSnapshot();
            });
        });
    });

    describe("not collapsible", () => {
        it("displays the content when the group is visible", () => {
            const accordionGroup = shallow(<AccordionGroup {...defaultAccordionGroupProps} collapsed={false} />);

            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(0);
            expect(accordionGroup).toMatchSnapshot();
        });

        it("displays the content when the group becomes visible", () => {
            const accordionGroup = shallow(
                <AccordionGroup {...defaultAccordionGroupProps} visible={false} collapsed={false} />
            );

            accordionGroup.setProps({ visible: true });
            expect(defaultAccordionGroupProps.generateHeaderIcon).toHaveBeenCalledTimes(0);
            expect(accordionGroup).toMatchSnapshot();
        });
    });

    it("render widgets when 'loadContent' is 'always'", () => {
        const accordionGroup = mount(
            <AccordionGroup
                {...defaultAccordionGroupProps}
                collapsible
                content={<div>Widgets</div>}
                loadContent={"always"}
            />
        );
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(1);
        expect(accordionGroup).toMatchSnapshot();
    });

    it("doesn't render widgets when 'loadContent' is 'whenExpanded'", () => {
        const accordionGroup = mount(
            <AccordionGroup
                {...defaultAccordionGroupProps}
                collapsible
                content={<div>Widgets</div>}
                loadContent={"whenExpanded"}
            />
        );
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(0);
        expect(accordionGroup).toMatchSnapshot();
    });

    it("render widgets when 'loadContent' is 'whenExpanded' and 'collapsed' is false", () => {
        const accordionGroup = mount(
            <AccordionGroup
                {...defaultAccordionGroupProps}
                collapsible
                content={<div>Widgets</div>}
                loadContent={"whenExpanded"}
                collapsed={false}
            />
        );
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(1);
    });

    it("keep widgets rendered when 'loadContent' is 'whenExpanded' and 'collapsed' prop becomes true", () => {
        const accordionGroup = mount(
            <AccordionGroup
                {...defaultAccordionGroupProps}
                collapsible
                content={<div>Widgets</div>}
                loadContent={"whenExpanded"}
                collapsed
            />
        );
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(0);

        accordionGroup.setProps({ collapsed: false });
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(1);

        accordionGroup.setProps({ collapsed: true });
        expect(accordionGroup.find({ children: "Widgets" })).toHaveLength(1);
    });
});
