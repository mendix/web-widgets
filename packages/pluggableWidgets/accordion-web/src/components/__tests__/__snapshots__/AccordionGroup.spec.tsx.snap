// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionGroup collapsible header doesn't render the icon when set to not visible 1`] = `
<section
  className="widget-accordion-group widget-accordion-group-collapsed class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={false}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group becomes visible and is collapsed 1`] = `
<section
  className="widget-accordion-group widget-accordion-group-collapsed class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={false}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group becomes visible and is expanded 1`] = `
<section
  className="widget-accordion-group class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={true}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group is visible and collapsed 1`] = `
<section
  className="widget-accordion-group widget-accordion-group-collapsed class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={false}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group is visible and expanded 1`] = `
<section
  className="widget-accordion-group class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={true}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group is visible and gets collapsed 1`] = `
<section
  className="widget-accordion-group widget-accordion-group-collapsed class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={false}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup collapsible renders correctly when the group is visible and gets expanded 1`] = `
<section
  className="widget-accordion-group class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={false}
      aria-expanded={true}
      className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
      data-focusindex={0}
      id="idHeaderButton"
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup doesn't render when the group isn't visible 1`] = `""`;

exports[`AccordionGroup doesn't render widgets when 'loadContent' is 'whenExpanded' 1`] = `
<AccordionGroup
  animateContent={false}
  collapsed={true}
  collapsible={true}
  content={
    <div>
      Widgets
    </div>
  }
  dynamicClassName="class-name"
  generateHeaderIcon={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  header="header"
  id="id"
  loadContent="whenExpanded"
  showHeaderIcon="right"
  visible={true}
>
  <section
    className="widget-accordion-group widget-accordion-group-collapsed class-name"
  >
    <header
      className="widget-accordion-group-header"
    >
      <div
        aria-controls="idContentWrapper"
        aria-disabled={false}
        aria-expanded={false}
        className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
        data-focusindex={0}
        id="idHeaderButton"
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        header
      </div>
    </header>
    <div
      aria-labelledby="idHeaderButton"
      className="widget-accordion-group-content-wrapper"
      data-focusindex={0}
      id="idContentWrapper"
      onTransitionEnd={[Function]}
      role="region"
    >
      <div
        className="widget-accordion-group-content"
      />
    </div>
  </section>
</AccordionGroup>
`;

exports[`AccordionGroup not collapsible displays the content when the group becomes visible 1`] = `
<section
  className="widget-accordion-group class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={true}
      aria-expanded={true}
      className="widget-accordion-group-header-button"
      data-focusindex={0}
      id="idHeaderButton"
      role="button"
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup not collapsible displays the content when the group is visible 1`] = `
<section
  className="widget-accordion-group class-name"
>
  <header
    className="widget-accordion-group-header"
  >
    <div
      aria-controls="idContentWrapper"
      aria-disabled={true}
      aria-expanded={true}
      className="widget-accordion-group-header-button"
      data-focusindex={0}
      id="idHeaderButton"
      role="button"
    >
      header
    </div>
  </header>
  <div
    aria-labelledby="idHeaderButton"
    className="widget-accordion-group-content-wrapper"
    data-focusindex={0}
    id="idContentWrapper"
    onTransitionEnd={[Function]}
    role="region"
  >
    <div
      className="widget-accordion-group-content"
    >
      <span>
        content
      </span>
    </div>
  </div>
</section>
`;

exports[`AccordionGroup render widgets when 'loadContent' is 'always' 1`] = `
<AccordionGroup
  animateContent={false}
  collapsed={true}
  collapsible={true}
  content={
    <div>
      Widgets
    </div>
  }
  dynamicClassName="class-name"
  generateHeaderIcon={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  header="header"
  id="id"
  loadContent="always"
  showHeaderIcon="right"
  visible={true}
>
  <section
    className="widget-accordion-group widget-accordion-group-collapsed class-name"
  >
    <header
      className="widget-accordion-group-header"
    >
      <div
        aria-controls="idContentWrapper"
        aria-disabled={false}
        aria-expanded={false}
        className="widget-accordion-group-header-button widget-accordion-group-header-button-clickable widget-accordion-group-header-button-icon-right"
        data-focusindex={0}
        id="idHeaderButton"
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        header
      </div>
    </header>
    <div
      aria-labelledby="idHeaderButton"
      className="widget-accordion-group-content-wrapper"
      data-focusindex={0}
      id="idContentWrapper"
      onTransitionEnd={[Function]}
      role="region"
    >
      <div
        className="widget-accordion-group-content"
      >
        <div>
          Widgets
        </div>
      </div>
    </div>
  </section>
</AccordionGroup>
`;
